Rename a few variables, e.g. `axis_steps_per_unit` into `axis_steps_per_mm`

gudnimg opened the pull request:
Proposal to rename a few variables to make the code more readable... also the names are the same as Marlin 2.1

List of renamed variables

| Current |  Proposal  |
|---|---|
| `axis_steps_per_unit`  | `axis_steps_per_mm`  |
| `axis_steps_per_sqr_second`  | `max_acceleration_steps_per_s2` |
| `max_acceleration_units_per_sq_second`  | `max_acceleration_mm_per_s2`  |
| `minsegmenttime`  | `min_segment_time_us`  |

Removed variables (dead code):
* `axis_travel_steps_per_sqr_second`
* `MAX_E_STEPS_PER_UNIT`
* `MIN_E_STEPS_PER_UNIT`

gudnimg committed:
Rename axis_steps_per_unit to axis_steps_per_mm

gudnimg committed:
Rename max_acceleration_units_per_sq_second to max_acceleration_mm_per_s2

gudnimg committed:
Change unsigned long to uint32_t

3d-gussner commented:
It is not only a rename PR with https://github.com/prusa3d/Prusa-Firmware/pull/4223/commits/eeb898a827c4d5f250bb0966cb07677cfbab333c

There are `AXIS_STEPS_PER_UNIT` upper case in the variants files and `ConfigurationStore.cpp` . 

`MAX_E_STEPS_PER_UNIT` and `MIN_E_STEPS_PER_UNIT` also found in variants and `ConfigurationStore.cpp` . 

Dead code around `axis_travel_steps_per_sqr_second`

gudnimg commented:
I’ll take a better look at your suggestions over the weekend. I need to be careful changing this many variables.

3d-gussner commented:
> I’ll take a better look at your suggestions over the weekend. I need to be careful changing this many variables.

If you like point out the variables you think we should change and I can go do that.

gudnimg committed:
Remove dead code axis_travel_steps_per_sqr_second

gudnimg committed:
Remove commented DEFAULT_AXIS_STEPS_PER_UNIT

They serve no function at this point

gudnimg commented:
Rebased branch to sync with MK3 branch.

gudnimg commented:
@3d-gussner feel free to change these variables if you have spare time next week :) I'll probably only focus on testing changes on the MMU FW side tomorrow.

* `DEFAULT_AXIS_STEPS_PER_UNIT` -->`DEFAULT_AXIS_STEPS_PER_MM`
* `MAX_E_STEPS_PER_UNIT` --> `MAX_E_STEPS_PER_MM`
* `MIN_E_STEPS_PER_UNIT` --> `MIN_E_STEPS_PER_MM`

I've removed the dead code surrounding `axis_travel_steps_per_sqr_second`

gudnimg committed:
Remove MAX_E_STEPS_PER_UNIT and MIN_E_STEPS_PER_UNIT

These constants are not used anywhere so there is no need to rename them
I propose they be removed as cleanup

gudnimg commented:
@3d-gussner I've added the following changes:
* **Renamed** `DEFAULT_AXIS_STEPS_PER_UNIT`  to `DEFAULT_AXIS_STEPS_PER_MM`
* **Removed** `MAX_E_STEPS_PER_UNIT` since it's not used anywhere.
* **Removed** `MIN_E_STEPS_PER_UNIT` since it's not used anywhere.

gudnimg committed:
Rename minsegmenttime --> min_segment_time_us

Also fixed wrong unit in Configuration prinout (M500)
should be us and not ms

Also used %lu format for uint32_t instead of %.2f

gudnimg committed:
fix build

gudnimg commented:
Sync with MK3 branch to resolve a merge conflict due to the power panic code being moved into a separate file.

gudnimg committed:
fixup

gudnimg commented:
* Changed `DEFAULT_AXIS_STEPS_PER_MM` back to `DEFAULT_AXIS_STEPS_PER_UNIT` to be consistent with Marlin 2.1
* Changed `DEFAULT_MIN_SEGMENT_TIME_US` back to `DEFAULT_MINSEGMENTTIME` to be consistent with Marlin 2.1

3d-gussner reviewed (approved):
Looks good to me and MK404 test print passed without any issues.

DRracer commented on Firmware/ConfigurationStore.h:
```diff
@@ -11,7 +11,7 @@ typedef struct
     float axis_steps_per_mm[4];
     float max_feedrate_normal[4];
-    unsigned long max_acceleration_mm_per_s2_normal[4];
+    uint32_t max_acceleration_mm_per_s2_normal[4];
```
:pray:

DRracer commented on Firmware/ConfigurationStore.cpp:
```diff
@@ -27,7 +27,7 @@ void Config_PrintSettings(uint8_t level)
 		"%SMaximum acceleration - stealth (mm/s2):\n%S  M201 X%lu Y%lu Z%lu E%lu\n"
 		"%SAcceleration: P=print, R=retract, T=travel\n%S  M204 P%.2f R%.2f T%.2f\n"
-		"%SAdvanced variables: S=Min feedrate (mm/s), T=Min travel feedrate (mm/s), B=minimum segment time (ms), X=maximum XY jerk (mm/s),  Z=maximum Z jerk (mm/s),  E=maximum E jerk (mm/s)\n%S  M205 S%.2f T%.2f B%.2f X%.2f Y%.2f Z%.2f E%.2f\n"
+		"%SAdvanced variables: S=Min feedrate (mm/s), T=Min travel feedrate (mm/s), B=minimum segment time (us), X=maximum XY jerk (mm/s),  Z=maximum Z jerk (mm/s),  E=maximum E jerk (mm/s)\n%S  M205 S%.2f T%.2f B%lu X%.2f Y%.2f Z%.2f E%.2f\n"
```
ideally, use `...%" PRIu32 "...`

gudnimg replied:
I didn't know about this macro :) 

If we use `PRIu32` here wouldn't it make sense to use it everywhere in this file for consistency?

DRracer replied:
it would be nice to do so, but not in this PR :wink: 
btw. there are more macros for printf like this...

DRracer commented on Firmware/variants/1_75mm_MK3-EINSy10a-E3DREVO.h:
```diff
@@ -38,7 +38,7 @@
 
 // Steps per unit {X,Y,Z,E}
-#define DEFAULT_AXIS_STEPS_PER_MM   {100,100,3200/8,280}
+#define DEFAULT_AXIS_STEPS_PER_UNIT   {100,100,3200/8,280}
```
Why are we changing `axis_steps_per_unit` into `axis_steps_per_mm` and `DEFAULT_AXIS_STEPS_PER_MM` into `DEFAULT_AXIS_STEPS_PER_UNIT`?

gudnimg replied:
It was due to a previous rebase to add the revo configs :( 

`DEFAULT_AXIS_STEPS_PER_UNIT` Is correct.

DRracer replied:
Don't we want this to revert to `DEFAULT_AXIS_STEPS_PER_MM` to keep it in sync with the rest of the renaming in this PR?

gudnimg replied:
I proposed to change `DEFAULT_AXIS_STEPS_PER_UNIT` to `DEFAULT_AXIS_STEPS_PER_MM` in this PR. But since Marlin 2.1 uses `DEFAULT_AXIS_STEPS_PER_UNIT` me and @3d-gussner agreed to be consistent with Marlin. 

But changing the variable `axis_steps_per_unit` into `axis_steps_per_mm` is consistent with Marlin 2.1

gudnimg replied:
The MK3 branch currently uses `DEFAULT_AXIS_STEPS_PER_UNIT` so I was just reverting a change made in this PR. The rebase made the history a bit confusing.

DRracer replied:
ok

DRracer reviewed (changes requested):
Please have a look at my comments, some are minor, but the DEFAULT_AXIS_STEPS_PER_UNIT is my biggest question :wink: .

3d-gussner commented:
> Please have a look at my comments, some are minor, but the DEFAULT_AXIS_STEPS_PER_UNIT is my biggest question 😉 .

We looked at Marlin 2.1 bugfix and Buddy firmware and there the `UNIT` stays as there might be someone using `inches` instead of `mm`. To keep 8- and 32-bit similar we reverted the some changes.
